azure/datalake/store/lib.py

```diff
@@ -282,35 +282,35 @@ def call(self, op, path='', **kwargs):
         func = getattr(requests, method)
         url = self.url + path
         try:
-            headers = self.head
+            headers = self.head.copy()
```

Comment:
I couldn't find other shared state or static members that would be impacted by concurrent calls.